main.tf

```diff
@@ -1,34 +1,102 @@
+### Default
 resource "aws_vpn_connection" "default" {
-  count = "${var.create_vpn_connection ? 1 : 0}"
+  count               = "${var.create_vpn_connection && length(var.tunnel1_inside_cidr) == 0  && length(var.tunnel2_inside_cidr) == 0 && length(var.tunnel1_preshared_key) == 0 && length(var.tunnel2_preshared_key) == 0 ? 1 : 0}"
 
   vpn_gateway_id      = "${var.vpn_gateway_id}"
   customer_gateway_id = "${var.customer_gateway_id}"
   type                = "ipsec.1"
 
-  static_routes_only = "${var.vpn_connection_static_routes_only}"
+  static_routes_only  = "${var.vpn_connection_static_routes_only}"
```

Comment:
Not related to your PR in general, but _while you are here_:
Add [this .pre-commit-config.yml](https://github.com/terraform-aws-modules/terraform-aws-security-group/blob/master/.pre-commit-config.yaml) to the root directory and run `pre-commit install && pre-commit run -a` to format all code properly automatically.